core/commands/commit.py

```diff
@@ -95,10 +96,21 @@ def run_async(self):
         if self.view.settings().get("git_savvy.commit_view.include_unstaged"):
             self.add_all_tracked_files()
 
+        commit_args = ["commit", "-q"]
+
         if self.view.settings().get("git_savvy.commit_view.amend"):
-            self.git("commit", "-q", "--amend", "-F", "-", stdin=commit_message)
-        else:
-            self.git("commit", "-q", "-F", "-", stdin=commit_message)
+            commit_args.append("--amend")
+
+        if self.view.settings().get("git_savvy.signed"):
+            commit_args.append("--gpg-sign")
+
+        commit_args.extend(["-F", "-"])
+
+        commit_kwargs = {
+            "stdin": commit_message
+        }
+
+        self.git(*commit_args, **commit_kwargs)
```

Comment:
Since you're not generating the keyword args dynamically, please change this line to:
```python
self.git(*commit_args, stdin=commit_message)
```